landa/water_body_management/doctype/catch_log_entry/catch_log_entry.py

```diff
@@ -20,3 +20,15 @@ def before_insert(self):
 			self.origin_of_catch_log_entry = "Regionalverband"
 		else:
 			self.origin_of_catch_log_entry = "Verein"
+
+	def validate(self):
+		tolerance = 0.4
+		for row in self.fish_catches:
+			typical_weight = frappe.db.get_value("Fish Species", row.fish_species, "typical_weight")
+			if (
+				row.weight_in_kg < typical_weight * (1 - tolerance) or
+				row.weight_in_kg > typical_weight * (1 + tolerance)
+				):
+				frappe.msgprint(
+                    frappe._("The weight of {0} in row {1} is not plausible").format(row.fish_species, row.idx)
+                )
```

Comment:
```suggestion
			if (
				row.weight_in_kg < typical_weight * (1 - tolerance) or
				row.weight_in_kg > typical_weight * (1 + tolerance)
			):
				frappe.msgprint(
					frappe._("The weight of {0} in row {1} is not plausible").format(row.fish_species, row.idx)
				)

```

In python ist whitespace wichtig